src/Text/Pandoc/App/CommandLineOptions.hs

```diff
@@ -529,9 +529,26 @@ options =
 
     , Option "" ["atx-headers"]
                  (NoArg
-                  (\opt -> return opt { optSetextHeaders = False } ))
+                  (\opt -> do
+                    deprecatedOption "--atx-headers"
+                        "This is now `--markdown-headings=atx`"
+                    return opt { optMarkdownHeadingFormat = HeadingFormatAtx } ))
                  "" -- "Use atx-style headers for markdown"
 
+    , Option "" ["markdown-headings"]
+                (ReqArg
+                  (\arg opt -> do
+                    headingFormat <-
+                      case arg of
+                        "setext" -> pure HeadingFormatSetext
+                        "atx" -> pure HeadingFormatAtx
+                        _ -> E.throwIO $ PandocOptionError $ T.pack
+                          ("Unknown markdown heading format: " ++ arg)
```

Comment:
Add: "expecting setext or atx"?

Reply:
I used your proposed message. Another one I thought was: "Valid options are: setext and atx". Both messages convey the same information.